src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsRecursion.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System.Xml.XPath;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen
+{
+    /// <summary>
+    ///     Service to handle recursive retrieval of XML comments from an XPathNavigator.
+    /// </summary>
+    internal static class XmlCommentsRecursionService
+    {
+        private const string MemberXPath = "/doc/members/member[@name='{0}']";
+
+        /// <summary>
+        ///     Finds the first node with the specified tag name recursively, starting from the given memberName in the XML
+        ///     document.
+        /// </summary>
+        /// <param name="xmlNavigator">The XPathNavigator representing the XML document.</param>
+        /// <param name="memberName">The name of the member to start the recursive search from.</param>
+        /// <param name="tag">The tag name to find.</param>
+        /// <returns>The XPathNavigator representing the found node or null if the node is not found.</returns>
+        private static XPathNavigator FindNodeRecursive(XPathNavigator xmlNavigator, string memberName, string tag)
+        {
+            while (true)
+            {
+                // Find the node representing the current memberName in the XML document.
+                var memberNode = xmlNavigator.SelectSingleNode(string.Format(MemberXPath, memberName));
+
+                // Try to find the specified tag node within the current member node.
+                var node = memberNode?.SelectSingleNode(tag);
+                if (node != null)
+                    return memberNode;
```

Comment:
Took me a second to understand what this was doing. Given `memberNode` might be `null` I think this would be more clearly written as:

```suggestion
                if (memberNode is null)
                {
                    return null;
                }

                // Try to find the specified tag node within the current member node.
                var node = memberNode.SelectSingleNode(tag);
                if (node != null)
                {
                    return memberNode;
                }
```